lib/order.rb

```diff
@@ -0,0 +1,72 @@
+require './lib/menu.rb'
+
+class Order
+
+	attr_reader :basket
+
+
+	def initialize (menu = Menu.new)
+		@menu = menu
+		@this_menu = {}
+		@basket = []
+		@confirmed_order = false
+		@item_costs = []
+
+	end
+
+	def this_menu
+		@this_menu = @menu.show_menu.dup
+	end
+
+	def add_to_basket(dish, quantity)
+		this_menu
+		raise 'Order already confirmed. Please submit new order' if confirmed_order?
+		raise 'Item not on menu' if !@this_menu.include?(dish)
```

Comment:
Style/NegatedIf: Favor `unless` over `if` for negative conditions. (https://github.com/bbatsov/ruby-style-guide#unless-for-negatives)<br>Style/SignalException: Use `fail` instead of `raise` to signal exceptions. (https://github.com/bbatsov/ruby-style-guide#fail-method)